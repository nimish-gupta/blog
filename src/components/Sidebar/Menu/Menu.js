// @flow strict
import React from 'react';
import { Link } from 'gatsby';
import styles from './Menu.module.scss';

type Props = {
	menu: {
		label: string,
		path?: string,
		href?: string,
	}[],
};

const Menu = ({ menu }: Props) =>
	console.log(menu) || (
		<nav className={styles['menu']}>
			<ul className={styles['menu__list']}>
				{menu.map((item) => (
					<li className={styles['menu__list-item']} key={item.path}>
						{item.path !== undefined ? (
							<Link
								to={item.path}
								className={styles['menu__list-item-link']}
								activeClassName={styles['menu__list-item-link--active']}>
								{item.label}
							</Link>
						) : (
							<a
								className={styles['menu__list-item-link']}
								target="_blank"
								rel="noreferer noopener"
								href={item.href}>
								{item.label}
							</a>
						)}
					</li>
				))}
			</ul>
		</nav>
	);

export default Menu;
