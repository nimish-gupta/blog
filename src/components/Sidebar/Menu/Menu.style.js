import styled from 'styled-components';
import { Link } from 'gatsby';

import { marginBottom } from '../../../assets/styled-components/mixins';

export const Menu = styled.nav`
	${marginBottom(1)};
`;

export const List = styled.ul`
	list-style: none;
	padding: 0;
	margin: 0;
`;

export const Item = styled.li`
	padding: 0;
	margin: 10px 0;
`;

const ItemNav = (comp) => (comp === 'a' ? styled.a : styled(comp))`
  font-size: ${(props) => props.theme.typographic.baseFontSize};
  color: ${(props) => props.theme.typographic.baseFontColor};
  font-weight: normal;
  border: 0;

  &:hover,
  &:focus {
    color: ${(props) => props.theme.color.primary};
    border-bottom: 1px solid ${(props) => props.theme.color.primary};
  }
  &.${(props) => props.activeClassName} {
    color: ${(props) => props.theme.color.base};
    border-bottom: 1px solid ${(props) => props.theme.color.base};
  }
`;

export const ItemAnchor = ItemNav('a');

export const ItemLink = ItemNav(Link);
