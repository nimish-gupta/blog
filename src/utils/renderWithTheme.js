import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from '../assets/styled-components/_variables';

export default function renderWithTheme(component) {
	return renderer.create(
		<ThemeProvider theme={theme}>{component}</ThemeProvider>
	);
}
