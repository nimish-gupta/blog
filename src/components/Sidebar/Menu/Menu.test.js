// @flow strict
import React from 'react';
import renderWithTheme from '../../../utils/renderWithTheme';
import Menu from './Menu';

describe('Menu', () => {
	const props = {
		menu: [
			{
				label: 'Item 0',
				path: '/#0/',
			},
			{
				label: 'Item 1',
				path: '/#1/',
			},
		],
	};

	it('renders correctly', () => {
		const tree = renderWithTheme(<Menu {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
