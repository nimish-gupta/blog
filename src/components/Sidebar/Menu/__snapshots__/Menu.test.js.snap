// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<nav
  className="sc-AxjAm jsZtex"
>
  <ul
    className="sc-AxirZ jsodwD"
  >
    <li
      className="sc-AxiKw fOyClS"
    >
      <a
        className="sc-AxhUy jXruOG"
        href="/#0/"
      >
        Item 0
      </a>
    </li>
    <li
      className="sc-AxiKw fOyClS"
    >
      <a
        className="sc-AxhUy jXruOG"
        href="/#1/"
      >
        Item 1
      </a>
    </li>
  </ul>
</nav>
`;
